laravel/resources/views/order.blade.php

```diff
@@ -47,14 +51,14 @@
             </tr>
             @foreach ($order_indexes as $order)
             <tr>
-                <td>{{$order->item_id}}</td>
+                <td>{{$order->item_code}}</td>
                 <td>{{$order->item->name}}</td>
-                <td>{{$order->delivery_date}}</td>
+                <td>{{$order->ship_date}}</td>
                 <td>{{$order->quantity}}</td>
                 <td>{{$order->name}}（仮）</td>
                 <td>{{$order->name}}（仮）</td>
-                <td>{{$order->delivery_user_id}}</td>
-                <td>{{$order->clientcompany->name}}</td>
+                <td>{{$order->order->delivery_user_id}}</td>
+                <td>{{$order->clientcompanydeliveryuser->name}}</td>
```

Comment:
ここはhasManyで定義しているfunction名の通り指定したほうが良さそうです。
```suggestion
                <td>{{$order-> clientCompanyDeliveryUser->name}}</td>
```

また `non-object` と言われてしまっているところを見る限り、 belongsToのリレーションで引っ張ってきている想定の `輸送先会社` のレコードが上手く取得できていないように思えます（null)

なので調査の流れとしては下記のようにやっていくと良さそうです
```
1. `$order->clientCompanyDeliveryUser` に値が入っているかを確認する
-> var_dumpなどを使いview側で変数が持っている情報を画面に表示してみる

2. もし値が入っていなかった場合、データベースのテーブルの中身を見て、delivery_user_idで指定しているidが `ClientCompany` のテーブルに存在するかを確認してみる
```